### PR TITLE
Fixed issue where large text on small space is truncated

### PR DESCRIPTION
**Pull request by madhavajay**
Font size search goes a few steps deeper and .Fit logic makes more sense
Binary search now goes exactly midway between each branch until termination
Tests updated to use specific optimalFontSize variable

I am not sure what you will think of this commit but the previous code was failing for me on some edge cases and this seems to solve them.

### CODE REVIEW

File: FittableFontLabel/FittableFontLabel iOS Tests/FittableFontLabelTests.swift [hunk 1]
```diff
@@ -24,6 +24,8 @@ class FittableFontLabelTests: XCTestCase {
     let shortText = "Etiam turpis elit"
     // swiftlint:disable:next line_length
     let longText = "Etiam turpis elit, efficitur nec suscipit vitae, aliquam vitae lorem. Aenean mauris tortor, semper vel facilisis nec, facilisis ut neque. Aliquam erat volutpat. Nam eget varius mauris, ut condimentum quam. Duis ex nisi, iaculis in ex ut, lobortis pretium est. Sed non lorem venenatis, fermentum quam aliquet, ultrices arcu. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Duis ullamcorper convallis tellus, vitae bibendum mi dapibus in. Morbi pellentesque aliquet urna, id aliquet orci efficitur sed. Donec et mi urna. Proin sagittis semper massa in semper."
+
+  let optimalFontSize: CGFloat = 2.0703125
```
Comment (tbaranes) [inline FittableFontLabel/FittableFontLabel iOS Tests/FittableFontLabelTests.swift]: Can you indent it at the same level as the others?

**Comment by tbaranes** [comment]
Yeah, I understand the issue here. Looks like a good solution to me! Thanks 👍 
If you can take a look at the comment, and add an entry in the CHANGELOG, then we will merge it in master

**Comment by madhavajay** [comment]
I just improved it actually, take another a look and let me know if you agree with the updated version, then ill update the CHANGELOG. 😎

**Comment by madhavajay** [comment]
I'm a bit confused by the versioning, the CHANGELOG only has Next, but the podspec in main already has 1.3.0. Ill leave that to you and just add the bug fix from these commits.

File: FittableFontLabel/FittableFontLabel iOS Tests/FittableFontLabelTests.swift (hunk 1 above)
Comment (madhavajay) [reply FittableFontLabel/FittableFontLabel iOS Tests/FittableFontLabelTests.swift]: It has been removed in the latest commit. I have now fixed the indentation of the subsequent commit as well.

**Comment by tbaranes** [comment]
Ups, my bad, I forgot to update the CHANGELOG after the last released 😬 
I will at  take a look at your change tonight 👍

**Comment by tbaranes** [comment]
Just merged it, and fix the CHANGELOG. Let me know you need your fix on master too, otherwise, it will be released at the same time as Swift3.

**Comment by madhavajay** [comment]
Awesome thanks. No rush im using my fork for now. 😎
